src/SegmentedArray.chpl

```diff
@@ -717,6 +717,78 @@ module SegmentedArray {
+      :arg pattern: pattern used to find matches of the leading and trailing characters in the segstring
+      :type pattern: string
+
+      :arg containsSpace: the string to replace pattern matches with
+      :type replStr: string
+
+      :arg initCount: If count is nonzero, at most count splits occur. If zero, substitute all occurences of pattern
+      :type initCount: int
+
+      :arg containsSpace: If True, appends a space character to the pattern
+      :type containsSpace: bool
+
+      :returns: Strings – substrings with stripped characters from the original string and the offsets into those substrings
+    */
+
+    proc strip(pattern: string, containsSpace: bool) throws {
+      checkCompile(pattern);
+      ref origOffsets = this.offsets.a;
+      ref origVals = this.values.a;
+      const lengths = this.getLengths();
+
+      saLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                                             "pattern: %s - origOffsets: %t - origVals: %t"
+                                             .format(pattern, origOffsets, origVals:bytes));
+
+      var replacedLens: [this.offsets.aD] int;
+      var newPattern: string;
+
+      // Pattern contains only whitespace characters or is empty
+      if containsSpace == true {
+        newPattern = pattern + " ";
+      }
+
+      saLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),"pattern: (%t)".format(newPattern));
+
+      forall (off, len, rlen) in zip(origOffsets, lengths, replacedLens) with (var valAgg = newDstAggregator(uint(8))) {
+        rlen = interpretAsBytes(origVals, off..#len).strip(newPattern:bytes).size + 1;
+      }
+      var retVals: [makeDistDom(+ reduce replacedLens)] uint(8);
+      var retOffs = (+ scan replacedLens) - replacedLens;
+
+      forall (off, len, roff) in zip(origOffsets, lengths, retOffs) with (var valAgg = newDstAggregator(uint(8))) {
+        var i = 0;
+        for b in interpretAsBytes(origVals, off..#len).strip(newPattern:bytes) {
+          valAgg.copy(retVals[roff+i], b:uint(8));
+          i += 1;
+        }
+      }
+      return (retOffs, retVals);
+
+      // Pattern contains only non-whitespace characters
+      // } else // if !pattern.isEmpty() {
+      //   forall (off, len, rlen) in zip(origOffsets, lengths, replacedLens) with (var valAgg = newDstAggregator(uint(8))) {
+      //     rlen = interpretAsBytes(origVals, off..#len).strip(pattern:bytes).size + 1;
+      //   }
+      //   var retVals: [makeDistDom(+ reduce replacedLens)] uint(8);
+      //   var retOffs = (+ scan replacedLens) - replacedLens;
+
+      //   forall (off, len, roff) in zip(origOffsets, lengths, retOffs) with (var valAgg = newDstAggregator(uint(8))) {
+      //     var i = 0;
+      //     for b in interpretAsBytes(origVals, off..#len).strip(pattern:bytes) {
+      //       valAgg.copy(retVals[roff+i], b:uint(8));
+      //       i += 1;
+      //     }
+      //   }
+      //   return (retOffs, retVals);
+    }
+
```

Comment:
If we aren't using this, it should be removed.

Reply:
Removed